power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -158,12 +158,39 @@ class Deserializer {
     std::vector<std::vector<ArraySpan>> msg_views_;
 
     static std::vector<char> json_to_msgpack(std::string_view json_string) {
-        nlohmann::json const json_document = nlohmann::json::parse(json_string);
+        nlohmann::json json_document = nlohmann::json::parse(json_string);
+        json_convert_inf(json_document);
         std::vector<char> msgpack_data;
         nlohmann::json::to_msgpack(json_document, msgpack_data);
         return msgpack_data;
     }
 
+    static void json_convert_inf(nlohmann::json& json_document) {
+        switch (json_document.type()) {
+        case nlohmann::json::value_t::object:
+        case nlohmann::json::value_t::array:
+            for (auto& value : json_document) {
+                json_convert_inf(value);
+            }
+            break;
+        case nlohmann::json::value_t::string:
```

Comment:
Sonar cloud

```suggestion
        using enum nlohmann::json::value_t;
        switch (json_document.type()) {
        case object:
        case array:
            for (auto& value : json_document) {
                json_convert_inf(value);
            }
            break;
        case string:
```

Reply:
You can't do that. `nlohmann::json::value_t` is a type alias to `nlohmann::detail::value_t`. `gcc/clang` does not allow you to `using enum` on a type alias, even it is aliasing a real enum class.